### PR TITLE
Add RestartSource to restart failing Kafka consumer for user-events service

### PR DESCRIPTION
**Pull request by upgle**
## Description

If there is a problem with the Kafka consumer, it no longer sends metrics. Because there is no error handling for user-events's Kafka consumer.

https://doc.akka.io/docs/alpakka-kafka/current/errorhandling.html

So I've added RestartSource to restart a failing consumer.

## Related issue and scope
- no related issue

## My changes affect the following components

-  User events

## Types of changes
- Enhancement or new feature (adds new functionality).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [x] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/EventConsumer.scala [hunk 1]
```diff
@@ -77,25 +77,34 @@ case class EventConsumer(settings: ConsumerSettings[String, String],
...
+
+  private val committerSettings = CommitterSettings(system)
+  private val control = new AtomicReference[Consumer.Control](Consumer.NoopControl)
+
+  private val result = RestartSource
+    .onFailuresWithBackoff(
+      minBackoff = metricConfig.retry.minBackoff,
+      maxBackoff = metricConfig.retry.maxBackoff,
+      randomFactor = metricConfig.retry.randomFactor,
+      maxRestarts = metricConfig.retry.maxRestarts) { () =>
```
Comment (chetanmeh) [inline core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/EventConsumer.scala:95]: May be add a logger which we can later check how many restart happened

File: core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/OpenWhiskEvents.scala [hunk 2]
```diff
@@ -29,11 +29,14 @@ import org.apache.kafka.common.serialization.StringDeserializer
...
 
+import scala.concurrent.duration.FiniteDuration
 import scala.concurrent.{ExecutionContext, Future}
 
 object OpenWhiskEvents extends SLF4JLogging {
 
-  case class MetricConfig(port: Int, enableKamon: Boolean, ignoredNamespaces: Set[String])
+  case class MetricConfig(port: Int, enableKamon: Boolean, ignoredNamespaces: Set[String], retry: RetryConfig)
+
+  case class RetryConfig(minBackoff: FiniteDuration, maxBackoff: FiniteDuration, randomFactor: Double, maxRestarts: Int)
```
Comment (upgle) [inline core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/OpenWhiskEvents.scala:39]: @chetanmeh Thank you for your review. I've used the same Configuration class here.   
https://github.com/apache/openwhisk/pull/4632/files#diff-3753d510c1485477c63054dde55b759dR39

Comment (upgle) [reply core/monitoring/user-events/src/main/scala/org/apache/openwhisk/core/monitoring/metrics/OpenWhiskEvents.scala:39]: @chetanmeh I'm sorry that I forgot to tell you that I referenced your code 😅

**Comment by style95** [comment]
@upgle 
Seems some configurations are missing for standalone openwhisk.
Could you look into this?

```
org.apache.openwhisk.core.monitoring.metrics.KamonRecorderTests > KamonConsumer should push user events to kamon FAILED
    pureconfig.error.ConfigReaderException: Cannot convert configuration to a org.apache.openwhisk.core.monitoring.metrics.OpenWhiskEvents$MetricConfig. Failures are:
      at 'user-events':
        - (file:/home/travis/build/apache/openwhisk/core/monitoring/user-events/build/resources/test/application.conf:18) Key not found: 'retry'.
        at pureconfig.package$.getResultOrThrow(package.scala:139)
        at pureconfig.package$.loadConfigOrThrow(package.scala:207)
        at org.apache.openwhisk.core.monitoring.metrics.EventsTestHelper.createConsumer(EventsTestHelper.scala:37)
        at org.apache.openwhisk.core.monitoring.metrics.EventsTestHelper.createConsumer$(EventsTestHelper.scala:31)
        at org.apache.openwhisk.core.monitoring.metrics.KafkaSpecBase.createConsumer(KafkaSpecBase.scala:28)
        at org.apache.openwhisk.core.monitoring.metrics.KamonRecorderTests.$anonfun$new$1(KamonRecorderTests.scala:72)
        at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
        at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
        at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
        at org.scalatest.Transformer.apply(Transformer.scala:22)
        at org.scalatest.Transformer.apply(Transformer.scala:20)
        at org.scalatest.FlatSpecLike$$anon$5.apply(FlatSpecLike.scala:1682)
        at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
        at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
        at org.apache.openwhisk.core.monitoring.metrics.KafkaSpecBase.withFixture(KafkaSpecBase.scala:28)
        at org.scalatest.FlatSpecLike.invokeWithFixture$1(FlatSpecLike.scala:1680)
        at org.scalatest.FlatSpecLike.$anonfun$runTest$1(FlatSpecLike.scala:1692)
        at org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
        at org.scalatest.FlatSpecLike.runTest(FlatSpecLike.scala:1692)
        at org.scalatest.FlatSpecLike.runTest$(FlatSpecLike.scala:1674)
        at org.apache.openwhisk.core.monitoring.metrics.KamonRecorderTests.org$scalatest$BeforeAndAfterEach$$super$runTest(KamonRecorderTests.scala:36)
        at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:221)
        at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:214)
        at org.apache.openwhisk.core.monitoring.metrics.KamonRecorderTests.runTest(KamonRecorderTests.scala:36)
```

**Comment by style95** [comment]
@upgle Thanks for addressing this.

**Pull request merged**